File: app/src/main/java/com/example/administrator/wuanandroid/ui/SeeWeekNews.java
package com.example.administrator.wuanandroid.ui;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import com.example.administrator.wuanandroid.Adapter.SeeWeekAdapter;
import com.example.administrator.wuanandroid.Bean.LoginBean.LoginRequestBean;
import com.example.administrator.wuanandroid.Bean.LoginBean.LoginResultBean;
import com.example.administrator.wuanandroid.Bean.SeeWeekBean.Request;
import com.example.administrator.wuanandroid.Bean.SeeWeekBean.Response;
import com.example.administrator.wuanandroid.Bean.SeeWeekBean.WeekNews;
import com.example.administrator.wuanandroid.R;
import com.example.administrator.wuanandroid.localAPI.PostRoute;
import com.example.administrator.wuanandroid.localAPI.SeeWeek;
import com.example.administrator.wuanandroid.utils.L;
import com.example.administrator.wuanandroid.utils.SharedUtil;
import com.google.gson.Gson;

import java.util.ArrayList;

import okhttp3.RequestBody;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

import static com.example.administrator.wuanandroid.utils.StaticClass.WUAN_URL;

public class SeeWeekNews extends AppCompatActivity {

    ArrayList<WeekNews> arrayList ;
    RecyclerView mRecyclerView;
    Button mSeeWeek;
    EditText mInput;
    String week;
    SharedUtil util = new SharedUtil();
    L l = new L();
    SeeWeekAdapter adapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_see_week_news);
        initView();



    }

    private void initView() {
        mRecyclerView = findViewById(R.id.seeWeek_RecyclerView);
        mSeeWeek = findViewById(R.id.to_See);
        mInput = findViewById(R.id.Seeweek_EditText);
        week = mInput.getText().toString();
        mSeeWeek.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                RequestToSeeWeekNews(); //查看周报的网络请求方法;
            }
        });
        LinearLayoutManager manager = new LinearLayoutManager(this);
        mRecyclerView.setLayoutManager(manager);
    }

    private void RequestToSeeWeekNews() {

        final Request request = new Request();
        StringBuffer route = new StringBuffer();
        request.setUserId(Integer.parseInt(util.getString(SeeWeekNews.this,"UserId","-1")));
        request.setWeekNum(Integer.parseInt(week));
        Gson gson = new Gson();
        route.append(gson.toJson(request));
        l.i(route.toString());

        Retrofit retrofit = new Retrofit.Builder()
                .baseUrl(WUAN_URL)
                .addConverterFactory(GsonConverterFactory.create())
                .build();

        SeeWeek seeWee = retrofit.create(SeeWeek.class);

        //创建请求体
        RequestBody body = RequestBody.create(okhttp3.MediaType.parse("application/json; charset=utf-8"),route.toString());
        Call<Response> call = seeWee.seeWeek(body);
        call.enqueue(new Callback<Response>() {
            @Override
            public void onResponse(Call<Response> call, retrofit2.Response<Response> response) {
                if (response.isSuccessful()){
                arrayList = new ArrayList<>();
                WeekNews weekNews = new WeekNews();
                weekNews.setTitle("本周完成");
                weekNews.setNeirong(response.body().getComplete().toString());
                arrayList.add(weekNews);
                WeekNews weekNews2 = new WeekNews();
                weekNews.setTitle("所遇困难");
                weekNews.setNeirong(response.body().getTrouble());
                arrayList.add(weekNews2);
                WeekNews weekNews3 = new WeekNews();
                weekNews.setTitle("下周完成");
                weekNews.setNeirong(response.body().getPlane());
                arrayList.add(weekNews3);
                WeekNews weekNews4 = new WeekNews();
                weekNews.setTitle("作品链接");
                weekNews.setNeirong(response.body().getUrl());
                arrayList.add(weekNews4);
                adapter = new SeeWeekAdapter(SeeWeekNews.this,arrayList);
                mRecyclerView.setAdapter(adapter);
                }else if(response.body().getInfoText().equals("未提交")){
                    arrayList = new ArrayList<>();
                    WeekNews weekNews = new WeekNews();
                    weekNews.setTitle("本周尚未提交周报");
                    weekNews.setNeirong(response.body().getComplete().toString());
                    adapter = new SeeWeekAdapter(SeeWeekNews.this,arrayList);
                    mRecyclerView.setAdapter(adapter);
                }else{
                    Toast.makeText(SeeWeekNews.this, "出现未知错误", Toast.LENGTH_SHORT).show();
                    l.i("500");
                }
            }

            @Override
            public void onFailure(Call<Response> call, Throwable t) {

            }
        });


    }
}

File: app/src/main/java/com/example/administrator/wuanandroid/ui/SeeWeekNews.kt
package com.example.administrator.wuanandroid.ui

import android.app.Dialog
import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v7.widget.StaggeredGridLayoutManager
import android.widget.LinearLayout
import android.widget.Toast
import com.example.administrator.wuanandroid.Adapter.MainRecyClickListen
import com.example.administrator.wuanandroid.Adapter.SeeAdapter
import com.example.administrator.wuanandroid.Bean.SeeWeekBean.SeeWeekReqeust
import com.example.administrator.wuanandroid.Bean.SeeWeekBean.SeeWeekResponse

import com.example.administrator.wuanandroid.R
import com.example.administrator.wuanandroid.utils.*
import com.google.gson.Gson
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers
import kotlinx.android.synthetic.main.activity_see_week_news.*
import okhttp3.RequestBody


class SeeWeekNews : AppCompatActivity() ,MainRecyClickListen{


    override fun MainRecyClickListener(bean: SeeWeekResponse.ReportsBean) {
        if(bean.status == 2){
            var i = Intent(this@SeeWeekNews,SeeNewsActivity::class.java)
            i.putExtra("1",bean.complete)
            i.putExtra("2",bean.trouble)
            i.putExtra("3",bean.plane)
            i.putExtra("4",bean.url)
            i.putExtra("5",bean.weekNum)

            startActivity(i)
        }else if(bean.status == 3 || bean.status == 1){
            var intent2 = Intent(this@SeeWeekNews,SeeweekleaveActivity::class.java)
            intent2.putExtra("status",bean.status)
            startActivity(Intent(intent2))
        }

    }


    var list  = ArrayList<SeeWeekResponse.ReportsBean>()

    var util = SharedUtil()
    var l = L()
    var t = ToastUtil(this@SeeWeekNews)
    var sp = StaggeredGridLayoutManager(2,LinearLayout.VERTICAL)
    var dialog:Dialog? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_see_week_news)
        see_back.setOnClickListener { finish() }
        dialog = DialogUtil(this@SeeWeekNews).getLoaddingDialog()

        RequestToSeeWeekNews()



    }




    private fun RequestToSeeWeekNews() {
        dialog!!.show()
        dialog!!.setCancelable(false)
        val request = SeeWeekReqeust()
        request.groupId = util.getInt(this@SeeWeekNews,StaticClass.GROUP_ID,1)
        request.pageNum = 1
        request.userId = util.getInt(this@SeeWeekNews,StaticClass.USER_ID,1)
        request.weekNum = util.getInt(this@SeeWeekNews,StaticClass.WEEK_NUM,1)
        val gson = Gson()
        var route = (gson.toJson(request))
        val body = RequestBody.create(okhttp3.MediaType.parse("application/json; charset=utf-8"), route)
        l.d(route)
        RequestUtil.observable
                .ISeeWeek(body)
                .observeOn(AndroidSchedulers.mainThread())
                .subscribeOn(Schedulers.io())
                .subscribe({
                    SeeWeekResponse->
                    dialog!!.dismiss()
                    l.d("${SeeWeekResponse.reports == null}")
                    if(SeeWeekResponse.reports != null){
                        for (item in SeeWeekResponse.reports!!){list.add(item) }
                        var adapter = SeeAdapter(this@SeeWeekNews,list!!,this)
                        see_recyview.layoutManager = sp
                        see_recyview.adapter = adapter
                    }else{
                        Toast.makeText(this@SeeWeekNews,"您还没有周报喔",Toast.LENGTH_SHORT)
                    }

                },{error->
                    dialog!!.dismiss()
                    Toast.makeText(this@SeeWeekNews,"服务器出现错误",Toast.LENGTH_SHORT)})
    }


}
